### PR TITLE
Add payment channel updated stream to ts rpc client

### LINKED ISSUE #1770: Add `PaymentChannelUpdated` event stream to TS rpc client
We already have this on the Go client, so it shouldn't be too much work.

### PR DESCRIPTION
**Pull request by geoknee**
Closes #1770 

- [x] Should we name it .onPaymentChannelUpdated ?

There is very little in the way of unit or integration tests setup at present. Therefore I have filed #1806, and we should make sure to cover this event stream in that work.

### CODE REVIEW

**Review by bitwiseguy — APPROVED** [review]

**Comment by geoknee** [comment]
@bitwiseguy any opinion on this 

>  Should we name it .onPaymentChannelUpdated ?

File: packages/nitro-rpc-client/src/interface.ts [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+  WaitForObjective(objectiveId: string): Promise<void>;
+  /**
+   * PaymentChannelUpdated attaches a callback which is triggered when the channel with supplied ID is updated.
+   *
+   * @param objectiveId - The id objective to wait for
+   */
+  PaymentChannelUpdated(
+    channelId: string,
+    callback: (info: PaymentChannelInfo) => void
+  ): Promise<void>;
```
Comment (geoknee) [inline packages/nitro-rpc-client/src/interface.ts:116]: Here's what I am wondering about the naming of.

Thinking about the usage, which of the following makes the most sense
```ts
nitroClient.PaymentChannelUpdated("0xabc",()=>{dothing()})
nitroClient.onPaymentChannelUpdated("0xabc",()=>{dothing()})
```

Comment (bitwiseguy) [reply packages/nitro-rpc-client/src/interface.ts:116]: Ok I got it. I don't have strong preference, but lean towards:
```go
nitroClient.onPaymentChannelUpdate("0xabc",()=>{dothing()})
```

Comment (geoknee) [reply packages/nitro-rpc-client/src/interface.ts]: [1b547d2](https://github.com/statechannels/go-nitro/pull/1803/commits/1b547d2a15d9c3c63cb87ce79c1cf6e57f946691)

**Pull request merged**